docs/installation/developers.rst

```diff
@@ -14,20 +16,29 @@ Once you have ``git`` in your system, clone the MUSE repository with:
 
     git clone https://github.com/EnergySystemsModellingLab/MUSE_OS.git
 
+For developing MUSE, **it is highly recommended to do so in a virtual Python environment**. This is to isolate dependencies and to avoid version conflicts on other Python libraries.
+Moreover, this allows for unconstrained experimentation with the code without breaking system dependencies, which, as least as far as ``Linux`` is concerned, is an issue.
 
-For developing MUSE, **it is highly recommended to use virtual environments**, so that you have more control on the versions of the dependencies being used, making easy to change them without affecting other tools using Python, or even to have several environments in parallel for different purposes (Eg. new features, bug fixing...). You can either use ``conda`` or ``venv`` as explained in :ref:`virtual-environments`.
+Also, any operating system allows for multiple Python versions being installed in parallel for different purposes like creating new features and bug fixing.
+You can either use ``conda`` or ``pyenv`` in conjunction with ``venv`` as explained in :ref:`virtual-environments`.
 
 Once you have your environment created, **activate it** and install MUSE within the environment:
 
 .. code-block:: bash
 
     cd MUSE_OS
-    # 1- Create virtual environment
-    # 2- Activate virtual environment
-    # 3- Finally, install MUSE in editable mode with:
-    python -m pip install -e .[dev,doc]
+    # 1- Create a virtual environment
+    # 2- Activate that virtual environment
+    # 3- Install MUSE in editable mode with: python -m pip install -e .[dev,doc]
+    # 4 - Invoke `muse --model default`
```

Comment:
So this is the key thing to enable all tests to pass when the full suite is run at once, right?

Reply:
I don't think so, as this will generate a results folder in the current working directory, whereas we specifically need it in the `docs` folder

Reply:
Yes. For the tests to pass without the failure one has to run `muse --model default`.
Independently, there are is still an "expected fail", some tests that got "skipped" and four "warnings", though I opened separate issues for them.

Reply:
> In the docs workflow [a link is made between both](https://github.com/EnergySystemsModellingLab/MUSE_OS/blob/d331f46c13a70889ca093492ea441848aec14a79/.github/workflows/documentation.yml#L29), but it is true I'm not entirely sure that's a solution in this case.

Yeah, that was a hack on my part to avoid having to generate results twice.

I think it's a good idea to tell users that they need to run this command in order for tests to pass, but I think we should put that in the "Running Tests" section, not here.